skimage/segmentation/manual_segmentation.py

```diff
@@ -0,0 +1,187 @@
+
+        elif event.button == 2:  # Middle click
+            if len(temp_list) > 1:
+
+                # Remove the previous vertice and update preview polygon
+                temp_list.remove(temp_list[-1])
+                preview_polygon_drawn[-1].remove()
+                preview_polygon_drawn.remove(preview_polygon_drawn[-1])
+
+                polygon = _draw_polygon(temp_list, alpha=alpha/1.2)
+                preview_polygon_drawn.append(polygon)
+
+            else:
+                return
+
+        elif event.button == 3:  # Right click
+            if len(temp_list) is 0:
+                return
+
+            # Store the vertices of the polygon as shown in preview.
+            # Redraw polygon and store it in polygons_drawn so that
+            # `_undo` works correctly.
+            list_of_verts.append(list(temp_list))
+            polygon_object = _draw_polygon(list_of_verts[-1])
+            polygons_drawn.append(polygon_object)
+
+            # Empty the temporary variables.
+            preview_polygon_drawn[-1].remove()
+            preview_polygon_drawn.remove(preview_polygon_drawn[-1])
+            del temp_list[:]
+
+            plt.draw()
+
+    def _on_select(verts):
+        list_of_verts.append(verts)
+        polygon_object = _draw_polygon(verts)
+        polygons_drawn.append(polygon_object)
+        plt.draw()
+
+    def _undo(event):
+        if len(list_of_verts) > 0:
+            list_of_verts.remove(list_of_verts[-1])
+
+            # Remove previous polygon object from the plot.
+            polygons_drawn[-1].remove()
+
+            # Removes latest polygon from a list of all polygon objects.
+            # This enables undo till the first drawn polygon.
+            polygons_drawn.remove(polygons_drawn[-1])
+
+        else:
+            return
+
+    image = np.squeeze(image)
+
+    if image.ndim not in (2, 3):
+        raise TypeError('Only 2-D images or 3-D images supported.')
+
+    fig, ax = plt.subplots()
+    ax.imshow(image)
+
+    buttonpos = plt.axes([0.85, 0.45, 0.075, 0.075])
+    undo_button = matplotlib.widgets.Button(buttonpos, u'\u27F2')
+
+    undo_button.on_clicked(_undo)
+
+    if draw_lines is True:
+        plt.suptitle(
+            "Left click to select vertex. Right click to confirm vertices.\
+        \nMiddle click to undo previous vertex.")
+
+        cid = fig.canvas.mpl_connect('button_press_event', _draw_lines)
+
+    else:
+        plt.suptitle("Draw around an object to select for mask generation.")
+        lasso = matplotlib.widgets.LassoSelector(ax, _on_select)
+
+    plt.show(block=True)
+
+    if return_masks is False:
+        mask = np.zeros(image.shape[:2], dtype=bool)
+        for verts in list_of_verts:
+            mask += _mask_from_verts(verts, image.shape[:2])
+    else:
+        mask = np.array(
+            [_mask_from_verts(verts,
+                              image.shape[:2]) for verts in list_of_verts])
+    return mask >= 1
```

Comment:
I think this unnecessarily throws out information. I would suggest just returning `mask`